### PR TITLE
add issue templates for promotion

### PR DESCRIPTION
**Pull request by crenshaw-dev**
I've created Reviewer and Approver promotion templates, since those are relatively easy to evaluate.

I'll leave adding a Lead template as a future task. That one's more difficult, because the requirements lend themselves less to a checklist-style template.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/promotion-reviewer.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+@<your-GH-handle>
+
+### Sub-project(s)
+
+Promotion is requested for the following sub-project(s).
+
+- [ ] Argo CD
+- [ ] Argo Events
+- [ ] Argo Rollouts
+- [ ] Argo Events
```
Comment (rohankmr414) [inline .github/ISSUE_TEMPLATE/promotion-reviewer.md:21]: typo?
```suggestion
- [ ] Argo Workflows
```

File: .github/ISSUE_TEMPLATE/promotion-approver.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+@<your-GH-handle>
+
+### Sub-project(s)
+
+Promotion is requested for the following sub-project(s). 
+
+- [ ] Argo CD
+- [ ] Argo Events
+- [ ] Argo Rollouts
+- [ ] Argo Events
```
Comment (rohankmr414) [inline .github/ISSUE_TEMPLATE/promotion-approver.md:21]: typo?
```suggestion
- [ ] Argo Workflows
```

**Review by rohankmr414 — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/promotion-approver.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
+name: Organization Approver Promotion Request
+about: Request promotion in a Argoproj Org
+title: 'REQUEST: Promotion to Approver for <your-GH-handle>'
+labels: area/github-membership
+assignees: ''
+
+---
+
```
Comment (crenshaw-dev) [inline .github/ISSUE_TEMPLATE/promotion-approver.md:9]:
```suggestion

<!--
IMPORTANT: A request for promotion to Approver should be initiated by an Approver or a Lead. After the Approver or Lead recommends promotion, the nominee should fill out this template and open an issue, listing the initiator as one of their sponsors.
-->

```

**Review by terrytangyuan — APPROVED** [review]
+1

**Review by sbose78 — APPROVED** [review]
Looks simple yet an effective way to propose a promotion.

**Pull request merged**